packages/common/src/content/HubContent.ts

```diff
@@ -1,90 +1,85 @@
-import { IItem } from "@esri/arcgis-rest-types";
-import { isMaster } from "cluster";
-import { fetchItemEnrichments } from "../items/_enrichments";
-import { getProp } from "../objects";
-import { getItemThumbnailUrl } from "../resources";
-import { IHubSearchResult } from "../search";
-import { parseInclude } from "../search/_internal/parseInclude";
-import { IHubRequestOptions } from "../types";
-import { getItemHomeUrl } from "../urls";
-import { unique } from "../util";
-import { mapBy } from "../utils";
-import { getFamily } from "./get-family";
-import { getHubRelativeUrl } from "./_internal";
+import { IArcGISContext } from "../ArcGISContext";
+import { HubItemEntity } from "../core/HubItemEntity";
+import { IHubEditableContent, IWithStoreBehavior } from "../core";
 
-/**
- * Enrich a generic search result
- * @param item
- * @param includes
- * @param requestOptions
- * @returns
- */
-export async function enrichContentSearchResult(
-  item: IItem,
-  include: string[],
-  requestOptions: IHubRequestOptions
-): Promise<IHubSearchResult> {
-  // Create the basic structure
-  const result: IHubSearchResult = {
-    access: item.access,
-    id: item.id,
-    type: item.type,
-    name: item.title,
-    owner: item.owner,
-    tags: item.tags,
-    typeKeywords: item.typeKeywords,
-    categories: item.categories,
-    summary: item.snippet || item.description,
-    createdDate: new Date(item.created),
-    createdDateSource: "item.created",
-    updatedDate: new Date(item.modified),
-    updatedDateSource: "item.modified",
-    family: getFamily(item.type),
-    links: {
-      self: "not-implemented",
-      siteRelative: "not-implemented",
-      thumbnail: "not-implemented",
-    },
-  };
+export class HubContent
+  extends HubItemEntity<IHubEditableContent>
+  implements IWithStoreBehavior<IHubEditableContent>
+{
+  private constructor(content: IHubEditableContent, context: IArcGISContext) {
+    super(content, context);
+  }
 
-  // default includes
-  const DEFAULTS: string[] = [];
+  /**
+   * Create an instance from an IHubEditableContent object
+   * @param json - JSON object to create a HubContent from
+   * @param context - ArcGIS context
+   * @returns
+   */
+  static fromJson(
+    json: Partial<IHubEditableContent>,
+    context: IArcGISContext
+  ): HubContent {
+    // TODO: merge what we have with the default values
+    // const pojo = this.applyDefaults(json, context);
+    const pojo = json as IHubEditableContent;
+    return new HubContent(pojo, context);
+  }
 
-  // Add any type-specific defaults here
-  // if (["Map Service", "Feature Service"].includes(item.type)) {
-  //   DEFAULTS = ["server.layers.length AS layerCount"];
-  // }
+  /**
+   * Save the HubContent to the backing store. Currently Projects are stored as Items in Portal
+   * @returns
+   */
+  async save(): Promise<void> {
+    this._checkDestroyed();
+    const { createContent, updateContent } = await import("./edit");
```

Comment:
I think it's cleaner if we just import these at the top of the file since this await and the two below are redundant.

Reply:
We do this so that the edit code is lazily loaded. Most users will only be viewing content (or any other entity). They should not have to wait for edit code to be downloaded and parsed before being able to interact w/ the application.

Another way to handle this would be to require that the entity classes are only `import`ed from other modules that are themselves lazily loaded (i.e. Stencil components), but that is not currently the case (we `import` the classes in Ember routes) and would be hard to enforce.

So for now at least, all entity classes should use dynamic `import()` for the edit code.

Reply:
Oh okay, that makes sense